packages/next/shared/lib/router/utils/parse-next-url.ts

```diff
@@ -0,0 +1,53 @@
+import { getCookieParser } from '../../../../server/api-utils'
+import { getLocaleMetadata } from '../../i18n/get-locale-metadata'
+import { parseUrl } from './parse-url'
+import type { NextConfig, DomainLocale } from '../../../../server/config-shared'
+import type { ParsedUrl } from './parse-url'
+import type { PathLocale } from '../../i18n/normalize-locale-path'
+
+interface Params {
+  headers?: { [key: string]: string | string[] | undefined }
+  nextConfig: NextConfig
+  url?: string
+}
+
+export function parseNextUrl({ headers, nextConfig, url = '/' }: Params) {
+  const urlParsed: ParsedNextUrl = parseUrl(url)
+  const { basePath } = nextConfig
```

Comment:
Why is only `basePath` parsed out here, do we want to parse out `i18n`, `trailingSlash`, and the other config items used here to de-dupe `nextConfig`?